napari/_qt/containers/qt_axis_model.py

```diff
@@ -0,0 +1,171 @@
+from typing import Any, Iterable, Union
+
+from qtpy.QtCore import QModelIndex, Qt
+
+from napari._qt.containers.qt_list_model import QtListModel
+from napari.components import Dims
+from napari.utils.events import SelectableEventedList
+
+
+class AxisModel:
+    """View of an axis within a dims model.
+
+    The model keeps track of axis names and allows read / write
+    acces on the cossesponding rollable state of a Dims object.
+
+    Parameters
+    ----------
+    dims : napari.components.dims.Dims
+        Parent Dims object.
+    axis : int
+        Axis index.
+
+    Attributes
+    ----------
+    dims : napari.components.dims.Dims
+        Dimensions object modeling slicing and displaying.
+    axis : int
+        Axis index.
+    """
+
+    def __init__(self, dims: Dims, axis: int) -> None:
+        self.dims = dims
+        self.axis = axis
+
+    def __hash__(self) -> int:
+        return id(self)
+
+    def __str__(self) -> str:
+        return repr(self)
+
+    def __repr__(self) -> str:
+        return self.dims.axis_labels[self.axis]
+
+    def __eq__(self, other: Union[int, str]) -> bool:
+        if not isinstance(other, AxisModel):
+            return NotImplemented
+        if isinstance(other, int):
+            return self.axis == other
+        return repr(self) == other
```

Comment:
The previous behavior seems to only care about comparing axis model instances with strings as indicated by the type annotation. That's a little weird because it doesn't actually handle other axis model instances, but I guess it's good enough for the internal usage. So, I think it's fine to leave the body/implementation of the function as before. Alternatively, you could add an extra type check if statement that evaluates `self.dims == other.dims and self.axis == other.axis` (or maybe `self.dims is other.dims`?).

```python
    def __eq__(self, other: object) -> bool:
        if isinstance(other, int):
            return self.axis == other
        if isinstance(other, str):
            return repr(self) == other
        if isinstance(other, AxisModel):
            return (self.dims is other.dims) and (self.axis == other.axis)
        return NotImplemented
```